io/src/ply/ply_parser.cpp

```diff
@@ -359,7 +359,10 @@ bool pcl::io::ply::ply_parser::parse (const std::string& filename)
             return false;
           }
         }
-        else if ((size_type_string == type_traits<uint32>::name ()) || (size_type_string == type_traits<uint32>::old_name ()))
+        // It is safe to use size_type = uint32 here even if it is actually int32, because the size/number of list entries is never negative,
+        // uint32 and int32 have the same width, and all allowed (non-negative) values have the same binary encoding in int32 and uint32.
```

Comment:
For a negative input, we can't detect it, but instead read it as unsigned int32, right?

Reply:
Yes, that is right. My considerations:
- If we wanted to treat int32 separately, not together with uint32, we would need a lot of additional logic and further changes, e.g. [in ply_parser.h](https://github.com/PointCloudLibrary/pcl/blob/master/io/include/pcl/io/ply/ply_parser.h#L196), in [ply_io.cpp](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/ply_io.cpp#L547), of course [in ply_parser.cpp](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/ply/ply_parser.cpp#L362), and maybe even more places I didn't find yet.
- A negative value for the number of elements would be a major mistake in the PLY file (this could only happen for binary files anyway, "-3" in ASCII could never be interpreted as an unsigned number). If we aim to detect such corruptions in the file, we would also have to check the upper bound (is e.g. 1 billion list entries still plausible, or likely a mistake?)

Overall, my view was that such a situation is highly unlikely, and that it doesn't justify the additional code.

Reply:
Agreed.